pkg/cmd/diagnostics/diagnostics_wait.go

```diff
@@ -0,0 +1,89 @@
+package diagnostics
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/dcos/dcos-cli/api"
+	diagnostics "github.com/dcos/dcos-core-cli/pkg/diagnostics/v2"
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
+	"github.com/spf13/cobra"
+)
+
+const pollInterval time.Duration = time.Second
+
+func newDiagnosticsWaitCommand(ctx api.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "wait <bundle-id>",
+		Short: "Wait until the given bundle is completed",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := pluginutil.HTTPClient("")
+			client := diagnostics.NewClient(c)
+
+			// this seemed to be the easist way to manage using two different methods
+			// of getting the right bundle depending on the arguments
+			bundle, err := func() (*diagnostics.Bundle, error) {
+				if len(args) == 0 {
+					return latestBundle(client)
+				}
+				return client.Get(args[0])
+			}()
+			if err != nil {
+				return err
+			}
+
+			// only print the ID after verifying that the bundle actually exists
+			fmt.Fprintln(ctx.Out(), bundle.ID)
+
+			// no need to poll if the bundle is done already
+			if bundle.IsFinished() {
+				return nil
+			}
+
+			for {
+				bundle, err = client.Get(bundle.ID)
```

Comment:
If this is moved at the end of the loop, the block above with `// no need to poll if the bundle is done already` can be removed.

Alternatively, it is also fine to keep the command simpler and not optimize here, eg. to only retrieve a bundleID at the beginning of the command (either through the arg or by fetching the latest bundle) and start polling from the loop.

Reply:
I'm not sure I understand your first suggestion.

Reply:
For the first, I meant:
``` go
for {
	if bundle.IsFinished() {
		break
	}

	time.Sleep(pollInterval)
	
	bundle, err = client.Get(bundle.ID)
	if err != nil {
		return err
	}			
}
```